internal/jimm/juju/migrationtarget.go

```diff
@@ -466,3 +489,68 @@ func (j *JujuManager) importModelFromDescription(ctx context.Context, targetCont
 	}
 	return nil
 }
+
+// CleanupPartialModelMigrations cleans up any partial model migrations that have exceeded the timeout.
+// It deletes the incoming model migration record, deletes the user mappings for the model,
+// and deletes the model record from JIMM's state.
+func (j *JujuManager) CleanupPartialModelMigrations(ctx context.Context) error {
+	const op = errors.Op("jimm.CleanupPartialModelMigrations")
+
+	// Get all incoming model migrations that have exceeded the timeout.
+	migrations, err := j.Database.GetIncomingModelMigrationsCreatedBefore(ctx, time.Now().Add(-TIMEOUT_PENDING_MIGRATION))
+	if err != nil {
+		return errors.E(op, fmt.Errorf("failed to get incoming model migrations: %w", err))
+	}
+	var errs []error
+	for _, migration := range migrations {
+		err := j.cleanupPartialModelMigration(ctx, migration)
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+	return goerr.Join(errs...)
+}
+
+// cleanupPartialModelMigration cleans up a partial model migration by deleting the incoming model migration record,
+// deleting the user mappings for the model, and deleting the model record from JIMM's state.
+func (j *JujuManager) cleanupPartialModelMigration(ctx context.Context, migration dbmodel.IncomingModelMigration) error {
+	const op = errors.Op("jimm.cleanupPartialModelMigration")
+
+	return j.Database.Transaction(func(db *db.Database) error {
+		// Delete the incoming model migration record.
+		err := j.Database.DeleteIncomingModelMigration(ctx, &migration)
+		if err != nil {
+			return errors.E(op, err)
+		}
+
+		// Delete user mappings for the model.
+		err = j.Database.DeleteUserMappingsByModelUUID(ctx, migration.ModelUUID.String)
+		if err != nil {
+			return errors.E(op, err)
+		}
+
+		// Delete the model record from JIMM's state.
+		// We don't need to check the migration-mode here, because we set the migration-mode to None
+		// in the same transaction as the deletion of the incoming model migration record.
+		// This ensures that there is a dangling incoming model migration record
+		// then the model can't have the migration-mode set to none.
```

Comment:
I think there is a missing "if" but even with it, this is very confusing to read and I mostly get it I think. I'm inclined towards just removing the entire comment. We understand that the model hasn't been successfully migrated if we still have an `incomingModelMigration` record (that in itself could replace the comment).
```suggestion
		// We don't need to check the migration-mode here, because we set the migration-mode to None
		// in the same transaction as the deletion of the incoming model migration record.
		// This ensures that if there is a dangling incoming model migration record
		// then the model can't have the migration-mode set to none.
```

Reply:
removed.